lib/endpoint/add-code-samples.js

```diff
@@ -38,232 +50,74 @@ function toShellExample ({ route, scope, baseUrl }) {
-    case 'orgs-create-hook':
-    case 'repos-create-hook':
-      return {
-        prepareParamDict: (paramDict) => {
-          delete paramDict['config.content_type']
-          delete paramDict['config.secret']
-          delete paramDict['config.insecure_ssl']
-        }
-      }
-    case 'orgs-update-hook':
-    case 'repos-update-hook':
-      return {
-        prepareParams: (params) => params.push('config'),
-        prepareParamDict: (paramDict) => {
-          delete paramDict.config
-          delete paramDict['config.content_type']
-          delete paramDict['config.secret']
-          delete paramDict['config.insecure_ssl']
-        }
-      }
-    case 'repos-add-protected-branch-required-status-checks-contexts':
-    case 'repos-remove-protected-branch-required-status-checks-contexts':
-    case 'repos-replace-protected-branch-required-status-checks-contexts':
-      return {
-        prepareParamDict: (paramDict) => {
-          Object.assign(paramDict, {
-            contexts: 'contexts'
-          })
-        }
-      }
-    case 'repos-add-protected-branch-team-restrictions':
-    case 'repos-remove-protected-branch-team-restrictions':
-    case 'repos-replace-protected-branch-team-restrictions':
-      return {
-        prepareParamDict: (paramDict) => {
-          Object.assign(paramDict, {
-            teams: 'teams'
-          })
-        }
-      }
-    case 'repos-add-protected-branch-user-restrictions':
-    case 'repos-remove-protected-branch-user-restrictions':
-    case 'repos-replace-protected-branch-user-restrictions':
-      return {
-        prepareParamDict: (paramDict) => {
-          Object.assign(paramDict, {
-            users: 'users'
-          })
-        }
-      }
-    case 'markdown-render-raw':
-      return {
-        prepareParamDict: (paramDict) => {
-          Object.assign(paramDict, {
-            data: 'data'
-          })
-        }
-      }
-    case 'enterprise-admin-create-global-hook':
-      return {
-        prepareParamDict: (paramDict) => {
-          delete paramDict['config.content_type']
-          delete paramDict['config.secret']
-          delete paramDict['config.insecure_ssl']
-        }
-      }
-    case 'enterprise-admin-update-global-hook':
-      return {
-        prepareParamDict: (paramDict) => {
-          Object.assign(paramDict, {
-            'config.url': 'config.url'
-          })
-        }
-      }
+function getExampleParamValue (name, schema) {
+  const value = PARAMETER_EXAMPLES[name]
+  if (value) {
+    return value
+  }
+  switch (schema.type) {
+    case 'string':
+      return name
+    case 'boolean':
+      return true
+    case 'integer':
+      return 42
+    case 'object':
+      return mapValues(
+        schema.properties,
+        (propSchema, propName) => getExampleParamValue(propName, propSchema)
+      )
+    case 'array':
+      return [ getExampleParamValue(name, schema.items) ]
   }
-  return {}
+  return '???' // unknown data type
```

Comment:
should we throw an error so the unknown data type is added instead of setting example value to `???`

Reply:
i thought about that...but because this is the result of scraping, throwing seemed a bit heavy-handed...it currently never gets there for `api.github.com`...i'm not necessarily opposed to throwing tho

Reply:
If it never gets there, let's throw. Because if we get there in future it will be most likely some kind of inconsistency with additions to the docs and we would catch it that way explicitly